initfiles/sbin/complete-uninstall.sh.in

```diff
@@ -100,21 +100,36 @@ if [ -n "${lpath}" -a -h "${lpath}" ] ; then
 fi
 
 if [ -e /etc/debian_version ]; then
-    echo "Removing DEB"
-    apt-get remove -y hpccsystems-platform
-    if [ $? -ne 0 ]; then
-       echo "An error has occured during removal.  Try \`apt-get -f remove hpccsystems-platform\` to fix this issue."
-       exit 1
+    pckname=$(apt-cache pkgnames hpccsystems-platform 2> /dev/null)
+    if [ -n "${pckname}" ] ; then
+        echo "Removing ${pckname}"
+        apt-get remove -y ${pckname}
+        dpkg -P ${pckname}
+        if [ $? -ne 0 ]; then
```

Comment:
I think you want to do the dpkg -P only if the apt-get remove was successful, so you need to move the dpkg cmd to after the if test.
```
apt-get remove -y ${pckname}
if [ $? -ne 0 ] ; then
    echo "Error ..."
    exit
fi
dpkg -P ${pckname}
```

Reply:
Not sure but maybe a single cmd of
```
apt-get purge ${pckname}
```
would work ?  Just a thought tho, I am not sure which is best

Reply:
I don't think there is any difference between dpkg and apt when it comes to uninstalling. They go through the same process of purging the package.

Reply:
Right, but $? is the status of the most recent command, so if the apt-get remove fails - do you want to issue the dpkg -P ?
If leaveenv is set on the command-line then we leave the config files - and does a purge remove those ?  If so we dont want to do a purge.
So lets go back to before - why do we need the purge command ?